db/migrate/20221223084601_add_archive_project_permission_to_project_admin_role.rb

```diff
@@ -0,0 +1,35 @@
+class AddArchiveProjectPermissionToProjectAdminRole < ActiveRecord::Migration[7.0]
+  def up
+    add_archive_project_permission_to_project_admin_role
+  end
+
+  def down
+    remove_archive_project_permission_from_all_roles
+  end
+
+  private
+
+  def add_archive_project_permission_to_project_admin_role
+    # Project admin role is one of the 5 initial roles created on seeding.
+    # It is the one with most permissions.
+    execute <<~SQL.squish
+      INSERT INTO
+      role_permissions
+      (permission, role_id, created_at, updated_at)
+        SELECT 'archive_project', role_id, NOW(), NOW()
+        FROM role_permissions
+        WHERE role_id <= 5
+        GROUP BY role_id
+        ORDER BY count(role_id) DESC
+        LIMIT 1
+    SQL
```

Comment:
I wouldn't add the permission automatically to existing roles unless we have a very good criteria of the roles we want to add the permission for. The amount of permission isn't a good criteria in my opinion since the seeded roles might have changed drastically. The way this is implemented now, the non member role might receive the permission.

In this case, I don't think that there is a solid criteria so I would leave as a manual task to add the permission to the correct roles.

Reply:
I was not sure if that was a good idea or not. I'll remove it.